Formula/geesefs.rb

```diff
@@ -0,0 +1,24 @@
+class Geesefs < Formula
+  desc "Finally, a good FUSE FS implementation over S3"
+  homepage "https://github.com/yandex-cloud/geesefs"
+  url "https://github.com/yandex-cloud/geesefs/archive/refs/heads/master.tar.gz"
+  sha256 "e6578adaffe47f9bf954847a79392b73ef118dc44a564e5d9bf61f5c8c509ce5"
+
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  # For tests
+  depends_on "bash" => :build
+  depends_on "python@3.10" => :build
+  depends_on "truncate" => :build
+  depends_on "openjdk" => :build
+
+  def install
+    system "go", "build", "-o", "#{bin}/geesefs"
+  end
+
+  test do
+    system "make", "run-test"
```

Comment:
We need a test that exercises the some of the functionality of the app.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Would it be fine to start s3proxy for testing? Geesefs needs an S3 server for testing